File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
    double square_of_sum(int number) {
        return pow((number * (number + 1) / 2), 2);
    }

    double sum_of_squares(int number) {
        return number * (number + 1) * (2 * number + 1) / 6;
    }

    double difference(int number) {
        return square_of_sum(number) - sum_of_squares(number);
    }
}  // namespace difference_of_squares

File: solutions/lua/difference-of-squares/1/difference-of-squares.lua
local function square_of_sum(number)
    return (number * (number+1) / 2)^2
end

local function sum_of_squares(number)
    return number * (number + 1) * (2 * number + 1) / 6
end

local function difference_of_squares(number)
    return square_of_sum(number) - sum_of_squares(number)
end

return {
  square_of_sum = square_of_sum,
  sum_of_squares = sum_of_squares,
  difference_of_squares = difference_of_squares
}
